### PR TITLE
add OCaml 5.0.0 compatibility

### LINKED ISSUE #19: OCaml 5.0.0 compatibility
The current `Dockerfile` approach requires two Dockerfiles, one for OCaml 4 and one for OCaml 5. (for installing dependencies. See: https://github.com/kxcteam/pubinfra.dockerfiles/pull/17/files)

Therefore, we want to use Docker's "ARG" feature to externally provide OCaml version information and switch installation scripts accordingly. (ref: [Understand how ARG and FROM interact](https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact) and [ARG](https://docs.docker.com/engine/reference/builder/#arg))

example:

```shell
$ docker build --build-arg package_install_script_filename=install_ocaml_4_pacakges.sh .
# In this case, we expect that `install_ocaml_4_pacakges.sh` will be called internally.


$ docker build --build-arg package_install_script_filename=install_ocaml_5_pacakges.sh .
# In this case, we expect that `install_ocaml_5_pacakges.sh` will be called internally.
```

### PR DESCRIPTION
**Pull request by kxc-wraikny**
resolve #19 

- [x] add script files to `/ocaml-general/install_ocaml_*_packages.sh`.
- [x] update `/ocaml-general/Dockerfile` to use above install script files.
- [x] enable ocaml-version 5.0.0 in `/.github/workflows/ocaml-general.yml`.
- [x] operation check
- [x] question: Should `js_of_ocaml-lwt.4.1.0` be installed with OCaml 4 as well?

### CODE REVIEW

File: ocaml-general/Dockerfile [hunk 1]
```diff
@@ -25,29 +25,11 @@ RUN apt-get update && apt-get install -y \
   opam
 
 ARG OCAML_VERSION=4.13.1
+ARG OCAML_VERSION_MAJOR=$(eval cut -d'.' -f1 <<< $OCAML_VERSION)
```
Comment (haochenx) [inline ocaml-general/Dockerfile:28]: https://github.com/kxcteam/pubinfra.dockerfiles/issues/19#issuecomment-1435736943 で issue の desc 編集していて、この arg は `package_install_script_filename` にしてほしい

そして具体的な値は CI Script のほうで計算したい

File: ocaml-general/install_ocaml_5_packages.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  ezjsonm.1.3.0 \
+  ppx_optcomp.v0.15.0 \
+  brr.0.0.4 \
+  prr.0.1.1 \
+  zed.3.2.0 \
+  tezt.3.0.0 \
+  ppx_inline_test.v0.15.0 \
+  alcotest.1.6.0 \
+  qcheck.0.20 \
+  qcheck-alcotest.0.20
```
Comment (haochenx) [inline ocaml-general/install_ocaml_5_packages.sh:22]:
```suggestion
  qcheck-alcotest.0.20 \
  && opam clean -y --logs --repo-cache --download-cache --switch-cleanup
```

File: ocaml-general/install_ocaml_4_packages.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+opam install --yes \
+  dune.3.4.0 \
+  merlin \
+  odoc.2.2.0 \
+  ppxlib.0.26.0 \
+  ppx_deriving.5.2.1 \
+  js_of_ocaml.4.1.0 js_of_ocaml-ppx.4.1.0 js_of_ocaml-ppx.4.1.0 \
```
Comment (haochenx) [inline ocaml-general/install_ocaml_4_packages.sh:9]:
```suggestion
  js_of_ocaml.4.1.0 js_of_ocaml-ppx.4.1.0 js_of_ocaml-lwt.4.1.0 \
```

File: .github/workflows/ocaml-general.yml [hunk 4]
```diff
@@ -36,6 +37,9 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
+      - name: Set up env var
+        run: echo "PACKAGE_INSTALL_SCRIPT_FILENAME=install_ocaml_$(eval cut -d'.' -f1 <<< ${{ matrix.ocaml-version }})_packages.sh" >> $GITHUB_ENV
```
Comment (haochenx) [inline .github/workflows/ocaml-general.yml:41]: `tee -a` を使ったほうがよいです

File: ocaml-general/Dockerfile [hunk 5]
```diff
@@ -25,29 +25,11 @@ RUN apt-get update && apt-get install -y \
   opam
 
 ARG OCAML_VERSION=4.13.1
```
Comment (haochenx) [inline ocaml-general/Dockerfile:27]:
```suggestion
ARG OCAML_VERSION=5.0.0
```
OCaml 5 対応したら default version は 5 にたい 🙏

File: ocaml-general/Dockerfile (hunk 1 above)
Comment (kxc-wraikny) [reply ocaml-general/Dockerfile]: CI: https://github.com/kxcteam/pubinfra.dockerfiles/pull/27/files#diff-a7b5da75a01c16adaee8ccb72257aed522273e9b2a6e5a6062f3b449c697ae4fR40-R71

Dockerfile: https://github.com/kxcteam/pubinfra.dockerfiles/pull/27/files#diff-26376760a5083e77d22a239902d6694967f302dac6ce62915c923d00f3614270R28

File: ocaml-general/install_ocaml_4_packages.sh (hunk 3 above)
Comment (kxc-wraikny) [reply ocaml-general/install_ocaml_4_packages.sh]: fixed https://github.com/kxcteam/pubinfra.dockerfiles/pull/27/files#diff-7c57c50eef83e6608a7914648bd7ad34df1bf616c49c0a4937ebdc5793347aa2R9

File: .github/workflows/ocaml-general.yml (hunk 4 above)
Comment (kxc-wraikny) [reply .github/workflows/ocaml-general.yml]: fixed https://github.com/kxcteam/pubinfra.dockerfiles/pull/27/files#diff-a7b5da75a01c16adaee8ccb72257aed522273e9b2a6e5a6062f3b449c697ae4fR41

File: ocaml-general/Dockerfile (hunk 5 above)
Comment (kxc-wraikny) [reply ocaml-general/Dockerfile]: https://github.com/kxcteam/pubinfra.dockerfiles/pull/27/commits/4e337a062c2c9b867298e042ed9050e8644c7a07

**Review by haochenx — APPROVED** [review]
LGTM. Let's merge!

**Pull request merged**